include/util/coordinate.hpp

```diff
@@ -181,6 +199,9 @@ bool operator==(const Coordinate lhs, const Coordinate rhs);
 bool operator==(const FloatCoordinate lhs, const FloatCoordinate rhs);
 std::ostream &operator<<(std::ostream &out, const Coordinate coordinate);
 std::ostream &operator<<(std::ostream &out, const FloatCoordinate coordinate);
+
+inline Coordinate::Coordinate(const FloatCoordinate &other) : Coordinate(toFixed(other.lon), toFixed(other.lat)) {}
```

Comment:
If you fwd declare `FloatCoordinate` you can put the function definition into the class where it belongs to and get the inlining for free.